iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -491,13 +491,11 @@ public static Expression generateExpressionFromPartitionSpec(Table table, Map<St
       String partColName = entry.getKey();
       if (partitionFieldMap.containsKey(partColName)) {
         PartitionField partitionField = partitionFieldMap.get(partColName);
-        Type resultType = partitionField.transform().getResultType(table.schema()
-            .findField(partitionField.sourceId()).type());
-        Object value = Conversions.fromPartitionString(resultType, entry.getValue());
-        TransformSpec.TransformType transformType = TransformSpec.fromString(partitionField.transform().toString());
-        Iterable<?> iterable = () -> Collections.singletonList(value).iterator();
-        if (TransformSpec.TransformType.IDENTITY == transformType) {
-          Expression boundPredicate = Expressions.in(partitionField.name(), iterable);
+        Transform<?, ?> transform = partitionField.transform();
+        if (transform.isIdentity()) {
+          Type resultType = transform.getResultType(table.schema().findField(partitionField.sourceId()).type());
```

Comment:
since it's identity no need for getResultType
```
final Type partKeyType = table.schema().findField(partitionField.sourceId()).type();
final Object value = Conversions.fromPartitionString(partKeyType, entry.getValue());
````

Reply:
please update also the SemanticException messages

- "Partition transforms are not supported here: %s", partColName));
- "No partition column by the name: %s", partColName));